src/backend/distributed/commands/table.c

```diff
@@ -182,6 +199,12 @@ PostprocessCreateTableStmt(CreateStmt *createStatement, const char *queryString)
 	{
 		ErrorOutForFKeyBetweenPostgresAndCitusLocalTable(relationId);
 	}
+
+	/* invalidate foreign key cache if the table involved in any foreign key */
```

Comment:
I think we should either add comment on why we only do it on PG 12 or move it outside of the ifdef. 

I saw that we have tests for local tables. Can we have tests for distributed tables? Maybe with partitioned distributed tables as they trigger auto distribution of tables?

Reply:
>I think we should either add comment on why we only do it on PG 12

Actually we already have a comment for this block, should I add some more right before `MarkInvalidateForeignKeyGraph` block ? 


>	/*
	 * Postgres processes foreign key constraints implied by CREATE TABLE
	 * commands by internally executing ALTER TABLE commands via standard
	 * process utility starting from PG13. Hence, we will already perform
	 * unsupported foreign key checks via PreprocessAlterTableStmt function
	 * in PG13. But for the older version, we need to do unsupported foreign
	 * key checks here.
	 */

Reply:
>I saw that we have tests for local tables. Can we have tests for distributed tables?

We actually have bunch of tests with citus tables too
https://github.com/citusdata/citus/pull/4453/files#diff-03e6ab2e83ba78346e5c55a2cd92fe9d06aaea3c71a92ccba0d46b0033e87041R313-R372

But I think I should add a few tests with partitioned tables

Reply:
> But I think I should add a few tests with partitioned & distributed tables


Yes, please do that. Thanks!

Reply:
added at 0a50e0ede08f97b16f4e9d31c2021490df2249ea & cc1ff00e8a16075157abe4dbc638a90dd98ec1b2